src/platforms/qurt/fc_addon/mpu_spi/mpu9x50_main.cpp

```diff
@@ -0,0 +1,667 @@
+static struct accel_report _accel;				/**< accel report */
+static struct mag_report _mag;						/**< mag report */
+static struct gyro_calibration_s _gyro_sc;				/**< gyro scale */
+static struct accel_calibration_s _accel_sc;			/**< accel scale */
+static struct mag_calibration_s _mag_sc;					/**< mag scale */
+static enum gyro_lpf_e _gyro_lpf = MPU9X50_GYRO_LPF_20HZ;	/**< gyro lpf enum value */
+static enum acc_lpf_e _accel_lpf = MPU9X50_ACC_LPF_20HZ;	/**< accel lpf enum value */
+static enum gyro_sample_rate_e _imu_sample_rate = MPU9x50_SAMPLE_RATE_500HZ;	/**< IMU sample rate enum */
+
+struct {
+	param_t accel_x_offset;
+	param_t accel_x_scale;
+	param_t accel_y_offset;
+	param_t accel_y_scale;
+	param_t accel_z_offset;
+	param_t accel_z_scale;
+	param_t gyro_x_offset;
+	param_t gyro_x_scale;
+	param_t gyro_y_offset;
+	param_t gyro_y_scale;
+	param_t gyro_z_offset;
+	param_t gyro_z_scale;
+	param_t mag_x_offset;
+	param_t mag_x_scale;
+	param_t mag_y_offset;
+	param_t mag_y_scale;
+	param_t mag_z_offset;
+	param_t mag_z_scale;
+	param_t gyro_lpf_enum;
+	param_t accel_lpf_enum;
+	param_t imu_sample_rate_enum;
+} _params_handles;  /**< parameter handles */
+
+bool exit_mreasurement = false;
+
+
+/** Print out the usage information */
+static void usage();
+
+/** mpu9x50 stop measurement */
+static void stop();
+
+/** task main trampoline function */
+static void	task_main_trampoline(int argc, char *argv[]);
+
+/** mpu9x50 measurement thread primary entry point */
+static void task_main(int argc, char *argv[]);
+
+/**
+ * create and advertise all publicatoins
+ * @return   true on success, false otherwise
+ */
+static bool create_pubs();
+
+/** update all sensor reports with the latest sensor reading */
+static void update_reports();
+
+/** publish all sensor reports */
+static void publish_reports();
+
+/** update all parameters */
+static void parameters_update();
+
+/** initialize all parameter handles and load the initial parameter values */
+static void parameters_init();
+
+/** poll parameter update */
+static void parameter_update_poll();
+
+static int64_t _isr_data_ready_timestamp = 0;
+
+/**
+ * MPU9x50 data ready interrupt service routine
+ * @param[out]	context	address of the context data provided by user whill
+ *							registering the interrupt servcie routine
+ */
+static void *data_ready_isr(void *context);
+
+void *data_ready_isr(void *context)
+{
+	if (exit_mreasurement) {
+		return NULL;
+	}
+
+	_isr_data_ready_timestamp = hrt_absolute_time();
+	// send signal to measurement thread
+	px4_task_kill(_task_handle, SIGRTMIN);
+
+	return NULL;
+}
+
+void parameter_update_poll()
+{
+	bool updated;
+
+	/* Check if parameters have changed */
+	orb_check(_params_sub, &updated);
+
+	if (updated) {
+		struct parameter_update_s param_update;
+		orb_copy(ORB_ID(parameter_update), _params_sub, &param_update);
+		parameters_update();
+	}
+}
+
+void parameters_update()
+{
+	PX4_DEBUG("mpu9x50_parameters_update");
```

Comment:
We are no longer using this driver in the fc_addon.  If you can accept this pull request, I will go ahead and delete the files.  

Regarding device ID's, it will be some time before we can support this change in muorb.  For now, I have modified the calibration code in a different PR to check for a qurt/posix build.